extensions/amp-story/1.0/test/test-amp-story-page.js

```diff
@@ -248,352 +219,431 @@ describes.realWin('amp-story-page', {amp: true}, (env) => {
+
+            page.setState(PageState.PLAYING);
+
+            // `setState` runs code that creates subtasks (Promise callbacks).
+            // Waits for the next frame to make sure all the subtasks are
+            // already executed when we run the assertions.
+            win.requestAnimationFrame(() => {
+              mediaPoolMock.verify();
+              done();
+            });
+          });
       });
-  });
+    });
 
-  it('should find pageIds in a goToPage action', async () => {
-    const actionButton = createElementWithAttributes(win.document, 'button', {
-      'id': 'actionButton',
-      'on': 'tap:story.goToPage(id=pageId)',
+    it('should build the background audio on layoutCallback', async () => {
+      element.setAttribute('background-audio', 'foo.mp3');
+      page.buildCallback();
+      await page.layoutCallback();
+      expect(
+        scopedQuerySelectorAll(element, Selectors.ALL_PLAYBACK_MEDIA)[0].tagName
+      ).to.equal('AUDIO');
     });
-    element.appendChild(actionButton);
-    page.buildCallback();
 
-    await page.layoutCallback();
-    const actions = page.actions_();
+    it('should register the background audio on layoutCallback', async () => {
+      element.setAttribute('background-audio', 'foo.mp3');
+      page.buildCallback();
+      const mediaPool = await page.mediaPoolPromise_;
+      const mediaPoolRegister = env.sandbox.stub(mediaPool, 'register');
+      await page.layoutCallback();
+
+      const audioEl = scopedQuerySelectorAll(
+        element,
+        Selectors.ALL_PLAYBACK_MEDIA
+      )[0];
+      expect(mediaPoolRegister).to.have.been.calledOnceWithExactly(audioEl);
+    });
 
-    expect(actions.length).to.be.equal(1);
-    expect(actions[0]).to.be.equal('pageId');
-  });
+    it('should register amp-audio on layoutCallback', async () => {
+      const ampAudioEl = win.document.createElement('amp-audio');
+      addAttributesToElement(ampAudioEl, {
+        'src': 'foo.mp3',
+        'layout': 'nodisplay',
+      });
 
-  it('should find pageIds in a goToPage action with multiple actions', async () => {
-    const multipleActionButton = createElementWithAttributes(
-      win.document,
-      'button',
-      {
-        'id': 'actionButton',
-        'on': 'tap:story.goToPage(id=pageId),foo.bar(baz=quux)',
-      }
-    );
-    element.appendChild(multipleActionButton);
-    page.buildCallback();
-
-    await page.layoutCallback();
-    const actions = page.actions_();
-
-    expect(actions.length).to.be.equal(1);
-    expect(actions[0]).to.be.equal('pageId');
-  });
-
-  it('should find pageIds in a goToPage action with multiple events', async () => {
-    const multipleEventsButton = createElementWithAttributes(
-      win.document,
-      'button',
-      {
+      page.element
+        .querySelector('amp-story-grid-layer')
+        .appendChild(ampAudioEl);
+      new AmpAudio(ampAudioEl);
+
+      ampAudioEl.build();
```

Comment:
This is done automatically when you're importing the extension as you did. If I recall correctly, you should not need to import AmpAudio.

Reply:
I tried that but couldn't get it to work :/. The test will timeout on `page.layoutCallback` because the audio element will never dispatch the `canplay` event, which only happens if we call the `build()` manually.

Reply:
We shouldn't need to add `amp-audio` to the extensions array then?

Reply:
We do, otherwise we'll get `TypeError: ampAudioEl.build is not a function`